File: kompas_maturalny_backend/app/graphql/mutations/users/sign_in_user.rb
module Mutations
  module Users
    class SignInUser < BaseMutation
      argument :email, String, required: true
      argument :password, String, required: true

      field :user, Types::UserType, null: true
      field :errors, [String], null: false

      def resolve(email: nil, password: nil)
        user = User.find_for_database_authentication(email: email)

        if user.present?
          if user.valid_password?(password)
            context[:current_user] = user
            {
                user: user,
                errors: []
            }
          else
            GraphQL::ExecutionError.new("Incorrect Email/Password")
          end
        else
            GraphQL::ExecutionError.new("User not registered on this application")
        end
      end
    end
  end
end

File: kompas_maturalny_web/graphql/mutations/sign_in_user.js
import gql from "graphql-tag";

export const SIGN_IN_USER = gql`
    mutation signInUser($email: String!, $password: String!){
        signInUser(input: {
            email: $email,
            password: $password
        }
        ) {
            user {
                id
                email
                role
                authenticationToken
            }
        }
    }
`;